whelktool/scripts/2018/11/cleanup-publication-duplicates.groovy

```diff
@@ -0,0 +1,67 @@
+import whelk.converter.marc.RestructPropertyValuesAndFlagStep
+
+merger = new RestructPropertyValuesAndFlagStep(
+    matchValuePattern: '^[0-9u]{4}$',
+    fuzzyMergeProperty: "date",
+    fuzzPattern: "[^0-9]+",
+    magicValueParts: ["[nu]": "[0-9nuxNUX?-]"]
+)
+
+boolean checkFuzzyDate(obj, target) {
+    def props =  'startYear' in obj ? ['startYear', 'endYear'] : ['year']
+    return merger.checkOnlyExpectedValuesInFuzzyProperty(target, obj, props)
+}
+
+selectBySqlWhere('''
+    data#>>'{@graph,1,publication}' LIKE '%"PrimaryPublication"%"Publication"%'
+    OR
+    data#>>'{@graph,1,publication}' LIKE '%"PrimaryPublication"%"PrimaryPublication"%'
+''') { data ->
+    def (record, instance, work) = data.graph
+
+    if (!isInstanceOf(instance, 'Instance')) {
+        return
+    }
+
+    List publications = instance.publication
+
+    if (!(publications instanceof List) || publications.size() < 2) {
```

Comment:
Doesn't this check need to be after the `removeAll` call below?

Reply:
Ah, no, it's just a guard to skip processing instances where there aren't multiple publications.

Reply:
Discussed offline, no change necessary.